tests/test-render.js

```diff
@@ -94,5 +94,19 @@ describe('Default Renderers', function() {
         //     console.log(item);
         // });
         //console.log(result['Keypr.ts']);
-    })
+    });
+
+    it('should render $ref properties', function() {
+        var testRef = fs.readFileSync(__dirname + '/apis/testref.yaml', 'UTF-8');
+        testRef = yaml.load(testRef);
+
+        data = process(testRef, {
+            moduleName: 'API.Client',
+            className: 'DarkApi',
+            type: 'angularTs'
+        });
+
+        renderFn = renderer.get('angularTs');
+        result = renderFn(data);
```

Comment:
You may consider trying to pattern match the generated line of code for the $ref line that is generated from this.